src/main/java/controllers/BookstoreRestController.java

```diff
@@ -0,0 +1,202 @@
+        if (bookstore == null)
+            return null;
+        Book book = new Book(name, isbn, picture, description, author, publisher);
+        bookstore.addBook(book);
+        book.setBookstore(bookstore);
+        bookRepository.save(book);
+        bookstoreRepository.save(bookstore);
+        return book;
+    }
+
+
+    //Customer REST endpoints
+    @GetMapping("/api/getCustomer")
+    public Customer getCustomer(@RequestParam(value = "id") long id) {
+        return customerRepository.findById(id);
+    }
+
+    @GetMapping("/api/getCustomers")
+    public Iterable<Customer> getCustomers() {
+        return customerRepository.findAll();
+    }
+
+    @PostMapping("/api/newCustomer")
+    public Customer newCustomer(@RequestParam(value = "name") String name, @RequestParam(value = "address") String address, @RequestParam(value = "email") String email, @RequestParam(value = "phoneNumber") String phoneNumber) {
+        Customer customer = new Customer(name, address, email, phoneNumber);
+        customerRepository.save(customer);
+        return customer;
+    }
+
+
+    //Shopping Cart REST endpoints
+    @GetMapping("/api/getShoppingCart")
+    public ShoppingCart getShoppingCart(@RequestParam(value = "id") long id) {
+        return shoppingCartRepository.findById(id);
+    }
+
+    @GetMapping("/api/getShoppingCarts")
+    public Iterable<ShoppingCart> getShoppingCarts() {
+        return shoppingCartRepository.findAll();
+    }
+
+    @GetMapping("/api/getShoppingCartsByCustomer")
+    public Iterable<ShoppingCart> getShoppingCartsByCustomer(@RequestParam(value = "customerID") long customerID) {
+        Customer customer = customerRepository.findById(customerID);
+        if (customer == null)
+            return null;
+        return shoppingCartRepository.findByCustomer(customer);
+    }
+
+    @PostMapping("/api/newShoppingCart")
+    public ShoppingCart newShoppingCart(@RequestParam(value = "customerID") long customerID) {
+        Customer customer = customerRepository.findById(customerID);
+        if (customer == null)
+            return null;
+        ShoppingCart shoppingCart = new ShoppingCart(customer);
+        shoppingCartRepository.save(shoppingCart);
+        customer.setShoppingCart(shoppingCart);
+        customerRepository.save(customer);
+        return shoppingCart;
+    }
+
+    @PostMapping("/api/addBookToShoppingCart")
+    public ShoppingCart addBookToShoppingCart(@RequestParam(value = "shoppingCartID") long shoppingCartID, @RequestParam(value = "bookID") long bookID) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(shoppingCartID);
+        if (shoppingCart == null)
+            return null;
+        Book book = bookRepository.findById(bookID);
+        if (book == null)
+            return null;
+        shoppingCart.addBook(book);
+        book.addShoppingCart(shoppingCart);
+        shoppingCartRepository.save(shoppingCart);
```

Comment:
Curious whether the shoppingCartRepository.save() would cascade to the book, if not, think this should this be updated in the model?

Reply:
Addressed in 2c97fc871504fda69602c799616c6dd02218960f